extensions/notebook/src/dialog/configurePython/configurePathPage.ts

```diff
@@ -0,0 +1,191 @@
+			this.instance.showErrorMessage(this.instance.InvalidLocationMsg);
+			return false;
+		}
+
+		this.model.pythonLocation = pythonLocation;
+		this.model.useExistingPython = !!this.existingInstallButton.checked;
+
+		return true;
+	}
+
+	private async updatePythonPathsDropdown(useExistingPython: boolean): Promise<void> {
+		await this.pythonDropdownLoader.updateProperties({ loading: true });
+		try {
+			let pythonPaths: PythonPathInfo[];
+			let dropdownValues: azdata.CategoryValue[];
+			if (useExistingPython) {
+				pythonPaths = await this.model.pythonPathsPromise;
+				if (pythonPaths && pythonPaths.length > 0) {
+					dropdownValues = pythonPaths.map(path => {
+						return {
+							displayName: `${path.installDir} (Python ${path.version})`,
+							name: path.installDir
+						};
+					});
+				} else {
+					dropdownValues = [{
+						displayName: 'No supported Python versions found.',
+						name: ''
+					}];
+				}
+			} else {
+				let defaultPath = JupyterServerInstallation.DefaultPythonLocation;
+				dropdownValues = [{
+					displayName: `${defaultPath} (Default)`,
+					name: defaultPath
+				}];
+			}
+
+			this.usingCustomPath = false;
+			await this.pythonLocationDropdown.updateProperties({
+				value: dropdownValues[0],
+				values: dropdownValues
+			});
+		} finally {
+			await this.pythonDropdownLoader.updateProperties({ loading: false });
+		}
+	}
+
+	private createInstallRadioButtons(modelBuilder: azdata.ModelBuilder, useExistingPython: boolean): void {
+		let buttonGroup = 'installationType';
+		this.newInstallButton = modelBuilder.radioButton()
+			.withProperties<azdata.RadioButtonProperties>({
+				name: buttonGroup,
+				label: localize('configurePython.newInstall', "New Python installation"),
+				checked: !useExistingPython
+			}).component();
+		this.newInstallButton.onDidClick(() => {
+			this.existingInstallButton.checked = false;
+			this.updatePythonPathsDropdown(false)
+				.catch(err => {
+					this.instance.showErrorMessage(utils.getErrorMessage(err));
+				});
+		});
+
+		this.existingInstallButton = modelBuilder.radioButton()
+			.withProperties<azdata.RadioButtonProperties>({
+				name: buttonGroup,
+				label: localize('configurePython.existingInstall', "Use existing Python installation"),
+				checked: useExistingPython
+			}).component();
+		this.existingInstallButton.onDidClick(() => {
+			this.newInstallButton.checked = false;
+			this.updatePythonPathsDropdown(true)
+				.catch(err => {
+					this.instance.showErrorMessage(utils.getErrorMessage(err));
+				});
+		});
+	}
+
+	private async handleBrowse(): Promise<void> {
+		let options: vscode.OpenDialogOptions = {
+			defaultUri: vscode.Uri.file(utils.getUserHome()),
+			canSelectFiles: false,
+			canSelectFolders: true,
+			canSelectMany: false,
+			openLabel: this.SelectFileLabel
+		};
+
+		let fileUris: vscode.Uri[] = await this.apiWrapper.showOpenDialog(options);
+		if (fileUris && fileUris[0]) {
```

Comment:
You can shorten checks like this : `fileUris?.[0]`

But you should also check the length - or does this return undefined if the user doesn't select anything?